src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -434,14 +434,24 @@ private SafeCurlHandle CreateRequestHandle(RequestCompletionSource state, GCHand
                 // Set maximum automatic redirection option
                 SetCurlOption(requestHandle, CURLoption.CURLOPT_MAXREDIRS, _maxAutomaticRedirections);
             }
-            if (state.RequestMessage.Content != null)
+
+            if (state.RequestMessage.Method == HttpMethod.Put)
             {
                 SetCurlOption(requestHandle, CURLoption.CURLOPT_UPLOAD, 1L);
             }
-            if (state.RequestMessage.Method == HttpMethod.Head)
+            else if (state.RequestMessage.Method == HttpMethod.Head)
             {
                 SetCurlOption(requestHandle, CURLoption.CURLOPT_NOBODY, 1L);
             }
+            else if (state.RequestMessage.Method == HttpMethod.Post)
```

Comment:
You think its better to use switch in this case where all "if"s in this method are not a part of a single condition ?